src/datasources/perf-ds/datasource.js

```diff
@@ -10,6 +10,7 @@ export class OpenNMSDatasource {
     this.name = instanceSettings.name;
     this.basicAuth = instanceSettings.basicAuth;
     this.withCredentials = instanceSettings.withCredentials;
+    this.interval = (instanceSettings.jsonData || {}).timeInterval;
```

Comment:
The variable `interval` is set, but doesn't appear to be used anywhere?

Reply:
'interval' isn't used directly, but referenced through 'intervalMs' instead. The calculation is [handled internally by Grafana](https://github.com/grafana/grafana/blob/25bcdbcab1b7e2329e6b147e4c505b29d00fb7a3/public/app/features/panel/metrics_panel_ctrl.ts#L157-L170).

They are also available to the users as  [$__interval and $__interval_ms](http://docs.grafana.org/reference/templating/#the-interval-variable) respectively.

Reply:
Right. I guess what I'm getting at is do we need this line?

Reply:
Yes, this is how we access the default value configured in the [datasource settings](https://github.com/OpenNMS/opennms-helm/pull/60/files#diff-ee658e695e923a4837d086d30aeb078bR10).

The default value overrides Grafana's own assumption about a suitable time interval, and then [calculateInterval()](https://github.com/grafana/grafana/blob/25bcdbcab1b7e2329e6b147e4c505b29d00fb7a3/public/app/features/panel/metrics_panel_ctrl.ts#L157-L170) *optionally* overrides the default value as well, with whatever the user specifies in the query options for a specific panel.

Without this line, Grafana will just make a guess on what it thinks the time interval should be, based on the selected time range. When looking at the "last 30 minutes", it figures 2 seconds is a suitable interval, which `org.opennms.newts.query.minimum_step` will then override.

`web.log` used to be filled with benign, but annoying, warnings like this before I added the default configuration:
```
2018-07-12 22:40:05,453 WARN  [qtp1059610547-58601] o.o.n.m.i.NewtsFetchStrategy: Requested step size 22500 is too small. Using 60000.
2018-07-12 22:42:44,862 WARN  [qtp1059610547-58601] o.o.n.m.i.NewtsFetchStrategy: Requested step size 22500 is too small. Using 60000.
2018-07-12 22:46:53,097 WARN  [qtp1059610547-59227] o.o.n.m.i.NewtsFetchStrategy: Requested step size 22500 is too small. Using 60000.
2018-07-12 22:47:10,444 WARN  [qtp1059610547-58453] o.o.n.m.i.NewtsFetchStrategy: Requested step size 5000 is too small. Using 60000.
```

Reply:
Got it. I had missed the fact that the `calculateInterval` call retrieved the interval via `this.panel.interval`, which references the variable in question.